src/ui/Analytics/LiveAnalyticsClient.ts

```diff
@@ -197,20 +204,26 @@ export class LiveAnalyticsClient implements IAnalyticsClient {
     this.pendingSearchAsYouTypeSearchEvent = new PendingSearchAsYouTypeSearchEvent(this.rootElement, this.endpoint, searchEvent, this.sendToCloud);
   }
 
-  private pushClickEvent(actionCause: IAnalyticsActionCause, metaObject: IChangeableAnalyticsMetaObject, result: IQueryResult, element: HTMLElement) {
+  private pushClickEvent(actionCause: IAnalyticsActionCause, metaObject: IChangeableAnalyticsMetaObject, result: IQueryResult, element: HTMLElement): Promise<any> {
     var event = this.buildClickEvent(actionCause, metaObject, result, element);
     this.checkToSendAnyPendingSearchAsYouType(actionCause);
     this.triggerChangeAnalyticsCustomData('ClickEvent', metaObject, event, { resultData: result });
     Assert.isNonEmptyString(event.searchQueryUid);
     Assert.isNonEmptyString(event.collectionName);
     Assert.isNonEmptyString(event.sourceName);
     Assert.isNumber(event.documentPosition);
-    Defer.defer(() => {
-      if (this.sendToCloud) {
-        this.endpoint.sendDocumentViewEvent(event);
-      }
-      $$(this.rootElement).trigger(AnalyticsEvents.documentViewEvent, {
-        documentViewEvent: APIAnalyticsBuilder.convertDocumentViewToAPI(event)
+    return new Promise((resolve) => {
```

Comment:
Same thing here : No defer, and return the `Promise` from the endpoint directly and do not create a new one.